sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -175,7 +175,13 @@ class ExpressionParserSuite extends AnalysisTest {
 
   test("like expressions") {
     assertEqual("a like 'pattern%'", 'a like "pattern%")
+    assertEqual("a like 'pattern%' escape '#'", 'a.like("pattern%", "#"))
+    assertEqual("a like 'pattern%' escape '\"'", 'a.like("pattern%", "\""))
+    intercept("a like 'pattern%' escape '##'", "Escape string must be empty or one character.")
     assertEqual("a not like 'pattern%'", !('a like "pattern%"))
+    assertEqual("a not like 'pattern%' escape '#'", !('a.like("pattern%", "#")))
+    assertEqual("a not like 'pattern%' escape '\"'", !('a.like("pattern%", "\"")))
+    intercept("a not like 'pattern%' escape '\"/'", "Escape string must be empty or one character.")
     assertEqual("a rlike 'pattern%'", 'a rlike "pattern%")
```

Comment:
Can you move these new assertions into a new test unit?

Reply:
OK. I will create new test unit.